vcd/metadata.go

```diff
@@ -321,9 +368,9 @@ func checkIgnoredMetadataConflicts(d *schema.ResourceData, vcdClient *VCDClient,
 	return nil
 }
 
-// updateMetadataInState updates metadata and metadata_entry in the Terraform state for the given receiver object.
+// updateMetadataInStateWithDeprecatedMetadataSupport updates metadata and metadata_entry in the Terraform state for the given receiver object.
 // This can be done as both are Computed, for compatibility reasons.
-func updateMetadataInState(d *schema.ResourceData, vcdClient *VCDClient, resourceType string, receiverObject metadataCompatible) diag.Diagnostics {
+func updateMetadataInStateWithDeprecatedMetadataSupport(d *schema.ResourceData, vcdClient *VCDClient, resourceType string, receiverObject metadataCompatible) diag.Diagnostics {
```

Comment:
I think the new name is a smart move. It will help when we do the 4.0 cleanup. That way, the longer name will disappear and the slender new name will stay

Reply:
Thanks, I probably missed something. Just to be clear, how will it disappear in 4.0?

Reply:
We will remove the `metadata` structure, leaving only `metadata_entry`. Then `updateMetadataInStateWithDeprecatedMetadataSupport` will not be needed anymore

Reply:
Yes, the idea is that everything (or most of the things) that need to be cleaned up or removed are centralized in `metadata.go`. I think `updateMetadataInStateWithDeprecatedMetadataSupport` helps to quickly see, in the future, that this function must be either changed or deleted (in this case, deleted).

It happens the same with `metadataEntryResourceSchemaWithDeprecatedSupport`, it should be removed eventually (In general, `metadata.go` will deserve a deep cleanup in v4.0 when it comes, as there are a lot of workarounds there).

I agree that leaving `updateMetadataInState` as it is (for old and new metadata) and creating a new `updateMetadataInStateWithoutDeprecatedSupport` (or similar) for only the new metadata would have created a cleaner PR, but it felt more natural for me to have the "odd"/"weird long name" in the function that must be eventually removed.

Reply:
Got it. Then just a final thought before resolving this conversation:

> I think updateMetadataInStateWithDeprecatedMetadataSupport helps to quickly see, in the future, that this function must be either changed or deleted (in this case, deleted).

If it's to be deleted, would a shorter `updateMetadataInStateDeprecated` name be sufficient? Or is it actually doing both deprecated and non-deprecated work, hence the current longer name (`...WithDeprecated...`) is needed?

Reply:
Yes, it updates old metadata, then the new one. I could shrink it to `updateMetadataInStateDeprecated` anyway, I think the important word is `Deprecated` so as long as it has it in the name, it should be good.

Reply:
> Yes, it updates old metadata, then the new one.

Thank you for making it clear! I'm OK with either func name now.